pkg/model/api/config/datasourcediscovery.go

```diff
@@ -27,6 +28,16 @@ type HTTPDiscovery struct {
 	config.RestConfigClient `json:",inline" yaml:",inline"`
 }
 
+func (d HTTPDiscovery) MarshalYAML() (interface{}, error) {
+	cfg := config.NewPublicRestConfigClient(&d.RestConfigClient)
+	return cfg, nil
+}
+
+func (d HTTPDiscovery) MarshalJSON() ([]byte, error) {
```

Comment:
Isn't it dangerous to do that? I mean never saved in db this ?

Reply:
the config is never saved in the database, so we should be good.

Besides in the config the usage of the type `secret.Hide` is working exactly like that: by overriding these methods

Reply:
sure sure. Ok if never saved then let's do that